src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversion.cs

```diff
@@ -634,6 +635,7 @@ public bool IsReference
         /// <remarks>
         /// Implicit and explicit user-defined conversions are described in section 6.4 of the C# language specification.
         /// </remarks>
+        [MemberNotNullWhen(true, nameof(Method), nameof(MethodSymbol))]
```

Comment:
I don't believe this is necessary. When construction occurs, these conditions are validated.

Reply:
I was a bit thrown off by the use of `MemberNotNullWhen` here too because this method doesn't initialize or check `Method` or `MethodSymbol`.
Adding the assertion would make sense to me, but I'm not sure it would hold true. 
Seems like we can create user-defined conversions that return `null` `Method`:
```
        internal Conversion(UserDefinedConversionResult conversionResult, bool isImplicit)
        {
            _kind = conversionResult.Kind == UserDefinedConversionResultKind.NoApplicableOperators
                ? ConversionKind.NoConversion
                : isImplicit ? ConversionKind.ImplicitUserDefined : ConversionKind.ExplicitUserDefined;

            _uncommonData = new UncommonData(
                isExtensionMethod: false,
                isArrayIndex: false,
                conversionResult: conversionResult,
                conversionMethod: null, // this is used for `Method`
                nestedConversions: default);
        }
```

---
In reply to: [599078854](https://github.com/dotnet/roslyn/pull/52009#discussion_r599078854) [](ancestors = 599078854)